exdb/tests/integration_tests.py

```diff
@@ -466,34 +468,51 @@ def test_number_of_experiences(self):
 
+    def test_does_not_get_drafts_when_hs_not_author(self):
+        self.create_experience('dr')
+        response = self.clients['hs'].get(reverse('home'))
+        self.assertEqual(len(response.context['experiences']), 0,
+                         "Hallstaff should not see drafts if they are not the author")
+
+    def test_gets_drafts_when_hs_is_author(self):
+        self.create_experience('dr', author=self.clients['hs'].user_object)
+        response = self.clients['hs'].get(reverse('home'))
+        self.assertEqual(len(response.context['experiences']), 1,
+                         "Hallstaff should be able to see their own drafts")
+
     @override_settings(HALLSTAFF_UPCOMING_TIMEDELTA=timedelta(days=7), RA_UPCOMING_TIMEDELTA=timedelta(days=0))
     def test_week_ahead(self):
-        self.create_experience('ad')
-        Experience.objects.get_or_create(author=self.clients['ra'].user_object,
-                                         name="E1", description="test description",
-                                         start_datetime=(now() + timedelta(days=2)),
-                                         end_datetime=(now() + timedelta(days=3)),
-                                         type=self.create_type(),
-                                         sub_type=self.create_sub_type(),
-                                         goal="Test Goal",
-                                         audience="b",
-                                         status="ad",
-                                         attendance=None,
-                                         next_approver=self.clients['hs'].user_object)
+        e1 = self.create_experience('ad')
+        e2 = Experience.objects.get_or_create(author=self.clients['ra'].user_object,
+                                              name="E1", description="test description",
+                                              start_datetime=(now() + timedelta(days=2)),
+                                              end_datetime=(now() + timedelta(days=3)),
+                                              type=self.create_type(),
+                                              sub_type=self.create_sub_type(),
+                                              goal="Test Goal",
+                                              audience="b",
+                                              status="ad",
+                                              attendance=None,
+                                              next_approver=self.clients['hs'].user_object)[0]
+        ExperienceApproval.objects.create(experience=e1, approver=self.clients['hs'].user_object)
+        ExperienceApproval.objects.create(experience=e2, approver=self.clients['hs'].user_object)
         response = self.clients['hs'].get(reverse('home'))
         self.assertEqual(len(response.context["upcoming"]), 1, "There should be 1 experience in the next week")
```

Comment:
It isn't immediately obvious what this test is checking for. Perhaps you should add a comment